### DISCUSSION TITLE
How can I define the containable ElementTypeIds for the root plane?

### DISCUSSION THREAD

**Discussion by rsoika** [eclipse-glsp/glsp#771] [Q&A] [answered]
I am using the method `BaseDiagramConfiguration.getShapeTypeHints()` to define the `ShapeTypeHint` for the different elements I have in my diagram. 

For example some elements can contain only specific other elements. I can define this by calling the `setContainableElementTypeIds` method for a specific element ShapeTypeHint:

```java
    ShapeTypeHint shapeTypeHint = new ShapeTypeHint("Container", true, true, false, true);
    shapeTypeHint.setContainableElementTypeIds(Arrays.asList(new String[] { "StartType","EndType" }););
```

But what I can't figure out is, how to define which of my elements are containable in the root plane? Currently every element can be placed to the root diagram, but I want to restrict the list to a subset. Is this possible?

Thanks for any hints.

**Comment by tortmayr** [comment, marked as answer]
Hi @rsoika,
you can also use a ShapeTypeHint to restrict the containable elements of the root plane (i.e. the graph element):

```ts
ShapeTypeHint rootTypeHint = new ShapeTypeHint(DefaultTypes.GRAPH, false, false, false, false);
      rootTypeHint.setContainableElementTypeIds(...);
```

**Reply by rsoika** [reply]
Thanks a lot! This works fine. I was not aware of the `DefaultTypes.GRAPH` and tried it first with my own DiagramType.
